tools/generators/pbxtargetdependencies/src/Generator/ConsolidateTargets.swift

```diff
@@ -140,11 +140,11 @@ extension Generator.ConsolidateTargets {
         func resolveDependency(
             _ depID: TargetID,
             for id: TargetID
-        ) throws -> PotentialConsolidatedTargetKey {
+        ) throws -> PotentialConsolidatedTargetKey? {
             guard let dependencyKey = targetIDMapping[depID] else {
-                throw PreconditionError(message: """
-Target "\(id)" dependency on "\(depID)" not found in `targetIDMapping`
-""")
+                // We don't throw here because the dependency was probably a
+                // "potential test host" that became unfocused
```

Comment:
I believe this is more performant than popping a depset on the Starlark side. I think I might be able to optimize this though, instead of having it done in 3 places.

Reply:
It's too much of a hassle to pre-filter it. Going to leave it like this.